src/main/scala/org/constellation/consensus/RoundManager.scala

```diff
@@ -30,23 +38,34 @@ class RoundManager(implicit dao: DAO) extends Actor with ActorLogging {
 
     case StartNewBlockCreationRound if !rounds.exists(_._2.startedByThisNode) =>
       createRoundData(dao).foreach { roundData =>
-        startRound(roundData, startedByThisNode = true)
-        context.parent ! NotifyFacilitators(roundData)
-        passToRoundActor(
-          LightTransactionsProposal(
-            roundData.roundId,
-            FacilitatorId(dao.id),
-            roundData.transactions.map(_.hash),
-            roundData.messages.map(_.signedMessageData.hash),
-            roundData.peers.flatMap(_.notification).toSeq))
-        log.debug(s"node: ${dao.id.short} starting new round: ${roundData.roundId}")
-        dao.blockFormationInProgress = true
+        resolveMissingParents(roundData.tipsSOE).onComplete {
+          case Failure(e) =>
+            log.error(e, s"unable to start block creation round due to: ${e.getMessage}")
+          case Success(_) =>
+            startRound(roundData, startedByThisNode = true)
+            context.parent ! NotifyFacilitators(roundData)
+            passToRoundActor(
+              LightTransactionsProposal(roundData.roundId,
+                                   FacilitatorId(dao.id),
+                                   roundData.transactions.map(_.hash),
+                                   roundData.messages.map(_.signedMessageData.hash),
+                                   roundData.peers.flatMap(_.notification).toSeq)
+            )
+            log.debug(s"node: ${dao.id.short} starting new round: ${roundData.roundId}")
+            dao.blockFormationInProgress = true
+        }
       }
 
     case cmd: ParticipateInBlockCreationRound =>
       log.debug(s"node: ${dao.id.short} participating in round: ${cmd.roundData.roundId}")
-      startRound(adjustPeers(cmd.roundData))
-      passToRoundActor(StartTransactionProposal(cmd.roundData.roundId))
+      resolveMissingParents(cmd.roundData.tipsSOE).onComplete {
+        case Failure(e) =>
+          log.error(e, s"unable to start block creation round due to: ${e.getMessage}")
+          // TODO: should I interrupt others and abort the whole Round ?
```

Comment:
@ryle-goehausen should I interrupt others and abort the whole Round ?

Reply:
Shouldn't necessarily interrupt the others but should inform them at the least. Consensus should be robust to individual facilitator failures.